File: BuildConfig.cpp
#include <fstream>
#include <nlohmann/json.hpp>

using json = nlohmann::json;
using namespace std;

int main(){
    map<string, vector<int>> mp;
    ifstream mi("mapping");
    ifstream js("subtasks.json");
    ofstream qj("config.json");
    json subs = json::parse(js), qu;
    js.close();
    string s;
    int x;
    while(mi>>s){
        mi >> x;
        mp[s].push_back(x);
    }
    mi.close();
    qu["packages"] = json::array();
    for(auto it = subs["subtasks"].begin(); it != subs["subtasks"].end(); ++it){
        s = it.key();
        qu["packages"].push_back({ {"score", it.value()["score"]}, {"tests", mp[s]} });
    }
    qj << qu.dump(4);
    qj.close();
    return 0;
}

File: build_config.py
import json

# open files
mp = {}
with open("mapping") as mi:
    for line in mi:
        s, x = line.strip().split()
        if s not in mp:
            mp[s] = []
        mp[s].append(int(x))

with open("subtasks.json") as js:
    subs = json.load(js)

# create config.json file
qu = {}
qu["packages"] = []
for key, value in subs["subtasks"].items():
    qu["packages"].append({"score": value["score"], "tests": mp[key]})

with open("config.json", "w") as qj:
    json.dump(qu, qj, indent=4)
